src/Jobs/DeleteDomains.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Stancl\Tenancy\Contracts\TenantWithDatabase;
+use Stancl\Tenancy\Database\Models\Domain;
+use Stancl\Tenancy\Events\DatabaseDeleted;
+use Stancl\Tenancy\Events\DeletingDatabase;
+use Stancl\Tenancy\Events\DeletingDomain;
+use Stancl\Tenancy\Events\DomainDeleted;
+
+class DeleteDomains
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /** @var TenantWithDatabase */
+    protected $tenant;
+
+    public function __construct(TenantWithDatabase $tenant)
+    {
+        $this->tenant = $tenant;
+    }
+
+    public function handle()
+    {
+        $this->tenant->domains()->delete();
```

Comment:
Does this trigger the `deleting` event for those domain models?

Reply:
> Does this trigger the deleting event for those domain models?

No. 
I think we need to get rid of bulk deletion and delete it using `domains->each->delete()` to make use of events.

Reply:
Correct. The current approach misses the point of why we are doing this. Please read the linked issue/comment in depth.